configs/seL4Config.cmake

```diff
@@ -42,16 +42,6 @@ macro(declare_seL4_arch sel4_arch)
         "x86;KernelArchX86;ARCH_X86;KernelSel4ArchX86_64 OR KernelSel4ArchIA32"
     )
 
-    # The following config options are legacy and can be removed if they
-    # aren't used anywhere anymore.
-    if(KernelArchARM)
```

Comment:
@axel-h, what are your thoughts about leaving behind a test for if these options are set and emitting a diagnostic informing about the deprecation?

Reply:
Will do this. I was not sure how much history this has, you might know this much better. Especially the "arm_hyp" looks like a big hack and seems aarch32 only?

Reply:
Most of the legacy is from when this was all Makefiles.  arm_hyp was needed as a new kernel architecture as back then the scripts that produced a kernel that was given to L4V could only handle changing the kernel architecture, so it was used to mean aarch32 + hyp extensions.  Now it would be possible to completely remove it and follow the same pattern as aarch64.  Just need to find a good time to pay the cost of deprecating the old config.